csrc/python_frontend/python_bindings.cpp

```diff
@@ -666,13 +669,17 @@ void initNvFuserPythonBindings(PyObject* module) {
             }
 
             Tensor out = self.defineTensor(sizes.size());
-            // TODO: replace computeContiguity with computeTensorDescriptor
+            std::vector<std::optional<bool>> contiguity;
+	    std::vector<int64_t> stride_order;
+	    std::tie(contiguity, stride_order) = computeTensorDescriptor(sizes, strides),
+	    
             self.defineRecord(new TensorRecord(
                 {self.recordingState(out())},
                 std::move(dim_sizes),
-                computeContiguity(sizes, strides),
+		contiguity,
                 dtype,
-                is_cpu));
+                is_cpu,
+		stride_order));
```

Comment:
NoteToMyself: The change here needs to be picked up by integration's cache logic.